autode/wrappers/MOPAC.py

```diff
@@ -50,97 +56,167 @@
-                fixed_atoms += calc.cartesian_constraints
+        # Distance in Å need to be converted to a0 and then the energy
+        # Ha e^-1 to kcal mol-1 e^-1
+        potentials.append(Constants.ha2kcalmol * Constants.a02ang * potential)
 
-            for i, atom in enumerate(atoms):
-                x, y, z = atom.coord
-                if i in fixed_atoms:
-                    print(f'{atom.label:<3}{x:^10.5f} 0 {y:^10.5f} 0 {z:^10.5f} 0', file=input_file)
-                else:
-                    print(f'{atom.label:<3}{x:^10.5f} 1 {y:^10.5f} 1 {z:^10.5f} 1', file=input_file)
+    with open(f'{calc.name}_mol.in', 'w') as pc_file:
+        print(f'\n{len(atoms)} 0', file=pc_file)
 
-        if calc.point_charges is not None:
-            potentials = []
-            for atom in atoms:
-                potential = 0
-                coord = atom.coord
-                for point_charge in calc.point_charges:
-                    potential += point_charge.charge / np.linalg.norm(coord - point_charge.coord)
+        for potential in potentials:
+            print(f'0 0 0 0 {potential}', file=pc_file)
 
-                # Distance in Å need to be converted to a0 and then the energy Ha e^-1 to kcal mol-1 e^-1
-                potentials.append(Constants.ha2kcalmol*Constants.a02ang*potential)
+    calc.additional_input_files.append(f'{calc.name}_mol.in')
+    return
 
-            with open(f'{calc.name}_mol.in', 'w') as pc_file:
-                print(f'\n{len(atoms)} 0', file=pc_file)
-                [print(f'0 0 0 0 {potential}', file=pc_file) for potential in potentials]
-            calc.additional_input_files.append(f'{calc.name}_mol.in')
 
+class MOPAC(ElectronicStructureMethod):
+
+    def generate_input(self, calc, molecule):
+
+        with open(calc.input.filename, 'w') as input_file:
+            keywords = get_keywords(calc.input, molecule)
+            print(*keywords, '\n\n', file=input_file)
+
+            atoms, fixed_atom_idxs = get_atoms_and_fixed_atom_indexes(molecule)
+
+            if molecule.constraints.cartesian is not None:
+                fixed_atom_idxs += calc.cartesian_constraints
+
+            print_atoms(input_file, atoms, fixed_atom_idxs)
+            print_point_charges(calc, atoms)
+
+        return None
+
+    def get_input_filename(self, calc):
+        return f'{calc.name}_mopac.mop'
+
+    def get_output_filename(self, calc):
+        return f'{calc.name}_mopac.out'
+
+    def execute(self, calc):
+
+        @work_in_tmp_dir(filenames_to_copy=calc.input.get_input_filenames(),
+                         kept_file_exts=('.mop', '.out'))
+        def execute_mopac():
+            logger.info(f'Setting the number of OMP threads to {calc.n_cores}')
+            os.environ['OMP_NUM_THREADS'] = str(calc.n_cores)
+            execute(calc, params=[calc.method.path, calc.input.filename])
+
+        execute_mopac()
         return None
 
+    def clean_up(self, calc):
+        pass
```

Comment:
??
    def clean_up(self, calc):
        os.remove(calc.input.filename)
        return None
??
actually, can't this be a method in the base class?

Reply:
Was thinking about removing  non just input files for specific codes, but I guess you could just override the base method. Will implement